### ISSUE TITLE
Unrecognized type with add_dependency_idls

### ISSUE THREAD

**Issue by csnardi** [web-platform-tests/wpt#11280]
In #11252 I used `add_dependency_idls()` for the DOM IDL test. I must've not tested with both the new and old version of the test somehow, because contrary to what I said, the number of tests actually did change by using `add_dependency_idls()`.

First, 5 tests were no longer generated:

```
WorkerGlobalScope interface: existence and properties of interface object
DedicatedWorkerGlobalScope interface: existence and properties of interface object 
SharedWorkerGlobalScope interface: existence and properties of interface object
WorkerNavigator interface: existence and properties of interface object
WorkerLocation interface: existence and properties of interface object
```

However, I checked, and these tests are already generated for the HTML IDL test, since each interface is defined in `html.idl` and not `dom.idl`.

Secondly, one additional test fails:

```
AbortSignal interface: new AbortController().signal must inherit property "onabort" with the proper type | Unrecognized type EventHandler
```

This seems like somehow `add_dependency_idls` missed `EventHandler`, as it is defined in `html.idl`.

So this is really two issues:

1. `add_untested_idls` seems to still allow some tests to be generated for interfaces that are defined in the untested IDLs.
2. `add_dependency_idls` missed adding a type somehow.

**Comment by gsnedders** [comment]
cc/ @lukebjerring

**Comment by lukebjerring** [comment]
AFK until next week but hopefully a general comment helps:
So, sadly, the order in which you call it matters, so you might need to swap dom and html depending on the inheritance. Worst case is a chain where you'd have to re-call add_dependency_idls on the same file again (note that the method should handle that OK).

If it missed a type, could be that it's not picking up a nested member correctly. The function is young and iterating - sorry for the hassle. @foolip also has context.

**Comment by foolip** [comment]
@csnardi can you try putting `add_dependency_idls` after `add_idls` and see if that makes any difference? I expect that it would, and can't quite understand why it works at all in the other order :)

**Comment by csnardi** [comment]
@foolip Yep, that does work in solving the failing test. The 5 tests are still missing, but as I mentioned above I think this is okay (and probably even more correct than the previous behavior). I'll upload a PR in a second.
